src/shared/dissect-image.c

```diff
@@ -1892,7 +1892,7 @@ int partition_pick_mount_options(
                 /* The ESP might contain a pre-boot random seed. Let's make this unaccessible to regular
                  * userspace. ESP/XBOOTLDR is almost certainly VFAT, hence if we don't know assume it is. */
                 if (!fstype || fstype_can_umask(fstype))
```

Comment:
fstype_can_umask checks for umask mount opt, while later fmask/dmask is used...

Reply:
Should I create a new `fstype_can_fmask_dmask` then? :thinking:

Reply:
Yeah. And given this is the only user of _umask, and for most use cases it's desirable to set separate fmask/dmask, I think it's sufficient to just remove _umask then

Reply:
I'll prefix this by saying that I know nothing about C. So here's my attempt at _something_ to see if I'm going in the right direction.